test/wasi/c/readdir.c

```diff
@@ -14,7 +14,7 @@ int main() {
 
   platform = getenv("NODE_PLATFORM");
   assert(platform != NULL);
-  has_d_type = (0 != strcmp(platform, "aix") && 0 != strcmp(platform, "sunos"));
+  has_d_type = (0 != strcmp(platform, "aix") && 0 != strcmp(platform, "sunos")) && 0 != strcmp(platform, "os400");
```

Comment:
CI failures are related to this change.

```suggestion
  has_d_type = (0 != strcmp(platform, "aix") && 0 != strcmp(platform, "sunos") && 0 != strcmp(platform, "os400"));
```

Reply:
ops look like I have a bracket out of place, fixing.

Reply:
@richardlau pushed a commit to fix.

Reply:
@mhdawson unfortunately that doesn't seemed to have fixed the test on AIX and SmartOS.

Reply:
see discussion below, does not seem to be related to the change itself, but changes in the tools used to compile the wasm versions of the tests.